### PR TITLE
Provide mode in symbolic notation for `restic ls --json`

### LINKED ISSUE #3542: Returns "permissions" on "ls" command
Output of `restic version`
--------------------------

0.12.1

What should restic do differently? Which functionality do you think we should add?
----------------------------------------------------------------------------------

restic ls with json output returns a "mode" property that is "Go" value. I need to have a standart property like "drwxr-w---" for example like the find command with a property called "permissions".

What are you trying to do? What problem would this solve?
---------------------------------------------------------

I am developing a web frontend and this app give ability to list snapshots, show files, etc. I am not able today to display the permissions of files/directories.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------

You love flowers :) Yes, I am developing a full backup app with a lot of features (repositories definitions, backups definitions, scheduling, watching, hooks, etc), but 95% of the job is made by restic. I investigated to find the backup of love but only restic has the flexibility I search. The world is better thanks to guys like you ! Congratulations !

Notes : original thread in https://forum.restic.net/t/restic-ls-understanding-file-mode-with-json/4371

### PR DESCRIPTION
**Pull request by magandrez**
This change adds "permissions" to the JSON output in symbolic notation, just as `restic find --json` does, and utilizing the same
naming.

What does this PR change? What problem does it solve?
-----------------------------------------------------
#3542

This change outputs permissions in symbolic notation for `restic ls --json` just as `restic find --json` does
aligned with the naming used in `find` command.

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------
Yes

https://forum.restic.net/t/restic-ls-understanding-file-mode-with-json/4371/6

closes #3542

Checklist
---------

- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added tests for all code changes.
- [ ] I have added documentation for relevant changes (in the manual).
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [x] I have run `gofmt` on the code in all commits.
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

### CODE REVIEW

**Review by MichaelEischer — COMMENTED** [review]
That changelog needs a small changes. Besides that the code seems fine, although I'd like to wait with merging until #3575 has been merged which should fix the test failure.

File: changelog/unreleased/issue-3542 [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Enhancement: Provide mode in symbolic notation for ls output in JSON
+
+Restic ls --json did not provide output in symbolic notation does e.g.
+restic find --json. Now it provides mode in symbolic notation aligned
+with restic find --json
+
+https://github.com/restic/restic/issues/3542
```
[the remark this answers is not in the document]
Comment (magandrez) [reply changelog/unreleased/issue-3542:6]: Done

File: changelog/unreleased/issue-3542 [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Enhancement: Provide mode in symbolic notation for ls output in JSON
```
[the remark this answers is not in the document]
Comment (magandrez) [reply changelog/unreleased/issue-3542]: Done.

File: changelog/unreleased/issue-3542 (hunk 1 above)
[the remark this answers is not in the document]
Comment (magandrez) [reply changelog/unreleased/issue-3542:6]: Done.

File: changelog/unreleased/issue-3542 [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+
+Restic ls --json did not provide output in symbolic notation does e.g.
```
[the remark this answers is not in the document]
Comment (magandrez) [reply changelog/unreleased/issue-3542]: Description re-written.

**Comment by magandrez** [comment]
@MichaelEischer this is ready from my side.

**Review by MichaelEischer — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**